### PR TITLE
Implement mitigation 4 of TM

### PR DESCRIPTION
**Pull request by hug-dev**
Adds trace logging whenever requests, responses, mappings or commands
are sent or received.

Signed-off-by: Hugues de Valon <hugues.devalon@arm.com>

### CODE REVIEW

File: src/front/front_end.rs [hunk 1]
```diff
@@ -56,35 +56,65 @@ impl FrontEndHandler {
...
+                }
+            };
+            let response = self.dispatcher.dispatch_request(request, app_name.clone());
+            trace!("dispatch_request egress");
+            response
+        };
+
+        // Serialise the response into bytes
         // Write bytes to stream
         match response.write_to_stream(&mut stream) {
```
Comment (ionut-arm) [inline src/front/front_end.rs:105]: Apologies for not asking this earlier, but why not just put a `trace!("dispatch_request egress");` just before this match? Or did you not want to write it if the `dispatcher` wasn't called?

File: src/front/front_end.rs [hunk 2]
```diff
@@ -56,35 +56,65 @@ impl FrontEndHandler {
...
+            if crate::utils::GlobalConfig::log_error_details() {
+                if let Some(app_name_string) = app_name.clone() {
+                    info!(
+                        "New request received from application name \"{}\"",
+                        app_name_string
+                    )
+                } else {
+                    info!("New request received without authentication")
+                }
+            };
```
Comment (ionut-arm) [inline src/front/front_end.rs:97]: Nice 👌

File: src/key_info_managers/on_disk_manager/mod.rs [hunk 3]
```diff
@@ -209,6 +206,12 @@ impl OnDiskKeyInfoManager {
                         ))?,
                     ) {
                         Ok(key_triple) => {
+                            if crate::utils::GlobalConfig::log_error_details() {
```
Comment (ionut-arm) [inline src/key_info_managers/on_disk_manager/mod.rs:209]: Maybe we should rename the flag to `log_details`

File: src/front/front_end.rs (hunk 1 above)
Comment (hug-dev) [reply src/front/front_end.rs:105]: Yes so that it is only written when the `dispatcher` has been called.

**Pull request merged**